packages/utils/odsp-doclib-utils/src/odspErrorUtils.ts

```diff
@@ -65,18 +65,46 @@ export function getSPOAndGraphRequestIdsFromResponse(headers: { get: (id: string
 
 export interface IFacetCodes {
     facetCodes?: string[];
- }
+}
 
-export function parseFacetCodes(response: string): string[] {
-    const stack: string[] = [];
-    let error;
-    try {
-        error = JSON.parse(response).error;
+/** Empirically-based model of error response inner error from ODSP */
+export interface OdspErrorResponseInnerError {
+    code?: string;
+    innerError?: OdspErrorResponseInnerError
+}
+
+/** Empirically-based model of error responses from ODSP */
+export interface OdspErrorResponse {
+    error: OdspErrorResponseInnerError & {
+        message: string;
     }
-    catch(e) {
-        return stack;
+}
+
+/** Empirically-based type guard for error responses from ODSP */
+function isOdspErrorResponse(x: any): x is OdspErrorResponse {
+    const error = x?.error;
+    return typeof(error?.message) === "string" &&
+        (error?.code === undefined || typeof(error?.code) === "string");
+}
+
+export function tryParseErrorResponse(
+    response: string | undefined,
+): { success: true, errorResponse: OdspErrorResponse } | { success: false } {
```

Comment:
It is much easier to work with  OdspErrorResponse  | undefined than with types like that.

Reply:
Yeah that's a fair critique. I'd consider cleaning up in a subsequent PR after my leave of absence.